app/controllers/application_modal_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ApplicationModalController < ApplicationController
```

Comment:
But I still can't see a proper reason to avoid Rails's MVC and existing controllers. Instead of creating helpers or partials to render everything, you put a lot of strange code to controllers. My question was like "is there any particular reason because of which you really want to break all the conventions and write all this subclassing stuff?"